gallery/tutorials/tutorials/micrograph_source.py

```diff
@@ -0,0 +1,220 @@
+"""
+=================
+Micrograph Source
+=================
+
+This tutorial will demonstrate how to set up and use ASPIRE's ``MicrographSource`` class.
+"""
+
+import numpy as np
+
+from aspire.image import Image
+from aspire.noise import WhiteNoiseAdder
+from aspire.operators import RadialCTFFilter
+from aspire.source import MicrographSource, Simulation
+
+# %%
+# Creating a Micrograph Source
+# ----------------------------
+# We need to generate our projections by creating a ``Simulation`` and passing it into our MicrographSource
+
+# Let's create our Simulation with a particle box size of 64 and one volume.
+n_particles_per_micrograph = 4
+n_micrographs = 4
+
+sim = Simulation(
+    L=64,
+    n=n_particles_per_micrograph * n_micrographs,
+    seed=1234,
+    C=1,
+    amplitudes=1,
+    offsets=0,
+)
+
+# %%
+# We'll pass our Simulation as an argument and give our MicrographSource other arguments.
+# In this example, our MicrographSource has 4 micrographs of size 500, each with 4 particles.
+micrograph = MicrographSource(
+    sim,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+    seed=1234,
+)
+
+# Plot the Micrographs
+micrograph.micrographs[:].show()
+
+# %%
+# CTF Filters
+# -----------
+# By default, no CTF corruption is configured. To apply CTF filters, we have to pass them as arguments to the Simulation
+
+# Create our list of CTF filters
+ctfs = [
+    RadialCTFFilter(pixel_size=4, voltage=200, defocus=15000, Cs=2.26, alpha=0.07, B=0),
+    RadialCTFFilter(pixel_size=4, voltage=200, defocus=10000, Cs=2.26, alpha=0.07, B=0),
+    RadialCTFFilter(pixel_size=4, voltage=200, defocus=20000, Cs=2.26, alpha=0.07, B=0),
+]
```

Comment:
Alright, I'll wait for input from Garrett. In the meantime, I'll have just one CTF in the list.

Reply:
This is a very good point. I'm wondering whether it may be better to feed the `MicrographSource` a volume and some angles (plus maybe amplitudes) since most of the other parts of `Simulation` are not applicable (shifts, CTFs, noise). We'd be repeating some of the `Simulation` code yes, but perhaps some of this can be factored out if necessary.

Reply:
Each particle can have a different ctf.

We usually rationalize just one ctf per micrograph when explaining things because it is simple.

I would avoid adding indexing goo to map per micrograph in the tutorial at this stage of the game. So the choice is either leave it as it was (a common pattern in our codebase), or just have one CTF in the list.  If the change to a single CTF has already been committed that's fine.